plugin/evm/warp/backend.go

```diff
@@ -0,0 +1,76 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package warp
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/cache"
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+	"github.com/ava-labs/avalanchego/vms/platformvm/teleporter"
+)
+
+var _ WarpBackend = &warpBackend{}
+
+// WarpBackend tracks signature eligible warp messages and provides an interface to fetch them.
+// The backend is also used to query for warp message signatures by the signature request handler.
+type WarpBackend interface {
+	// AddMessage signs [unsignedMessage] and adds it to the warp backend database
+	AddMessage(ctx context.Context, unsignedMessage *teleporter.UnsignedMessage) error
+
+	// GetSignature returns the signature of the requested message hash.
+	GetSignature(ctx context.Context, messageHash ids.ID) ([]byte, error)
+}
+
+// warpBackend implements WarpBackend, keeps track of warp messages, and generates message signatures.
+type warpBackend struct {
+	db             database.Database
+	snowCtx        *snow.Context
+	signatureCache *cache.LRU
+}
+
+// NewWarpBackend creates a new WarpBackend, and initializes the signature cache and message tracking database.
+func NewWarpBackend(snowCtx *snow.Context, db database.Database, signatureCacheSize int) WarpBackend {
+	return &warpBackend{
+		db:             db,
```

Comment:
should we use a prefixed db here?

Reply:
we opted to pass in a `database.Database`, assuming the passed in db will be prefixed. https://github.com/ava-labs/subnet-evm/pull/452#discussion_r1084524857